src/control/functor/multivariate.lean

```diff
@@ -0,0 +1,770 @@
+/-
+Copyright (c) 2018 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jeremy Avigad, Mario Carneiro, Simon Hudon
+-/
+import data.fin2
+import logic.function.basic
+import tactic.basic
+
+/-!
+
+Tuples of types, and their categorical structure.
+
+Features:
+
+`typevec n`   : n-tuples of types
+`α ⟹ β`      : n-tuples of maps
+`f ⊚ g`       : composition
```

Comment:
I think it would be good to split this file into two files: one for typevec and the other for mvfunctor.  I don't think there are any dependencies from typevec on mvfunctor.